packages/legacy/core/App/services/keychain.ts

```diff
@@ -142,22 +143,30 @@ export const loadWalletKey = async (title?: string, description?: string): Promi
 export const loadWalletSecret = async (
   title?: string,
   description?: string
-): Promise<{ secret: WalletSecret | undefined; err: string }> => {
+): Promise<WalletSecret | undefined> => {
   let salt: WalletSalt | undefined
   let key: WalletKey | undefined
-  let err = ''
+  let secret: WalletSecret | undefined
   try {
     salt = await loadWalletSalt()
     key = await loadWalletKey(title, description)
   } catch (e: any) {
-    err = e?.message ?? e
+    // should this throw instead?
+    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, e?.message ?? e)
   }
 
-  return { secret: { ...salt, ...key } as WalletSecret, err }
+  if (salt && key) {
```

Comment:
Should we do some kind of validation that the `salt` has all the required properties?